### PR TITLE
MCD train; CSV ingester row count

### PR DESCRIPTION
**Pull request by kexinrong**
- Pass in a copy of input data to MCD so that the order of the original input is maintained
- Fix CSV ingester total row count

### CODE REVIEW

File: legacy/src/main/java/macrobase/ingest/CSVIngester.java [hunk 1]
```diff
@@ -103,7 +103,7 @@ private Datum parseRecord(CSVRecord record) throws NumberFormatException {
                     badRows++;
                 }
             }
-            log.info("{}/{} rows successfully parsed ({} malformed rows)", numRows-badRows, numRows, badRows);
```
Comment (edgan8) [inline legacy/src/main/java/macrobase/ingest/CSVIngester.java]: lol nice catch. Was this actually causing issues?

File: legacy/src/main/java/macrobase/analysis/transform/BatchScoreFeatureTransform.java [hunk 2]
```diff
@@ -39,8 +40,11 @@ public void initialize() {
 
     @Override
     public void consume(List<Datum> records) {
-        if(requiresTraining)
-            batchTrainScore.train(records);
+        if(requiresTraining) {
+            List<Datum> copy = new ArrayList<>(records);
```
Comment (edgan8) [inline legacy/src/main/java/macrobase/analysis/transform/BatchScoreFeatureTransform.java]: Not every featuretransform is going to modify its records. Can we add the copy to MCD so other transforms don't need to pay the overhead?

File: legacy/src/main/java/macrobase/ingest/CSVIngester.java (hunk 1 above)
Comment (kexinrong) [reply legacy/src/main/java/macrobase/ingest/CSVIngester.java]: Not really. I just noticed that the line count numbers were wrong.

File: legacy/src/main/java/macrobase/analysis/transform/BatchScoreFeatureTransform.java (hunk 2 above)
Comment (kexinrong) [reply legacy/src/main/java/macrobase/analysis/transform/BatchScoreFeatureTransform.java]: Make sense. Will do.

**Pull request merged**